analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/PublicMethodArgumentsShouldBeCheckedForNull.CSharp9.cs

```diff
@@ -23,29 +23,29 @@ public void PatternMatching(object arg)
 
         if (arg is int and > 0 and > 1)
         {
-            arg.ToString();     // Compliant
+            arg.ToString();     // Noncompliant - FP
         }
 
         if (arg is int or bool or long)
         {
-            arg.ToString();     // Compliant
+            arg.ToString();     // Noncompliant - FP
         }
 
         if (arg is null)
         {
-            arg.ToString();     // FN
+            arg.ToString();     // Covered by S2259
         }
         if (arg is int or bool or null)
         {
-            arg.ToString();     // FN
+            arg.ToString();
         }
         else if (arg is not not null)
         {
-            arg.ToString();     // FN
+            arg.ToString();
         }
         else if (!(arg is not null))
         {
-            arg.ToString();     // FN
+            arg.ToString();
         }
```

Comment:
These should still be `FN`.

Reply:
The null check was performed on the parameters before they were used, so aren't these cases belong to S2259?